File: internal/cli/label/usage.go
package label

const usageText = `lin label — Search, list, and inspect Linear labels (issue or project)

SUBCOMMANDS:
  label list [--type <t>] [--team <team>] [--name <text>] [--is-group[=false]]   List labels (filterable)
  label search <text> [--type <t>] [--team <team>]                               Substring search (case- and accent-insensitive)
  label get <id|name> [--type <t>] [--team <team>]                               Single label by UUID or exact name

OPTIONS:
  --type <t>        Label type: "issue" (default) or "project"
  --team <team>     Filter/disambiguate by team key, name, or UUID (issue labels only)
  --name <text>     Exact match (case-insensitive)
  --is-group        Only group labels (--is-group=false for non-groups)
  --limit <n>       Limit results (list, search)
  --cursor <token>  Pagination cursor (list, search)

OUTPUT FIELDS:
  list/search (issue)   → id, name, color, [description, isGroup, team{id,key,name}, parent{id,name}]
  list/search (project) → id, name, color, [description, isGroup, parent{id,name}]
  get                   → same fields, single object

NOTES:
  Issue labels can be workspace-wide or team-scoped; --team filters/disambiguates.
  Project labels are workspace-only — --team is rejected with --type=project.
  IssueLabel and ProjectLabel are distinct Linear entities; the same name can exist in both.
  Use the resulting issue label name (with --team) or UUID with "issue new --labels" / "issue update labels".
  Use a project label name or UUID with "project update labels".`

File: bun/src/cli/user/usage.ts
import type { Command } from "commander";

const USAGE_TEXT = `lin user — Look up Linear workspace users

SUBCOMMANDS:
  user search <text>         Search users by name, email, or display name
  user list [--team <team>]  List users (optionally filtered by team)
  user me                    Current authenticated user + organization

OPTIONS (search):
  --limit <n>               Limit results
  --cursor <token>          Pagination cursor for next page

OPTIONS (list):
  --team <team>             Filter by team key or name (e.g., "ENG")
  --limit <n>               Limit results
  --cursor <token>          Pagination cursor for next page

OUTPUT FIELDS:
  search → id, name, email, displayName
  list   → id, name, email, displayName
  me     → id, name, email, displayName, organization (id, name)

NOTES:
  User IDs are UUIDs. Use "user me" to get the current user's ID for
  operations like --assignee filtering.
  When --team is specified, returns only members of that team.
  Search matches against name, email, and displayName (case-insensitive).
`;

export function registerUsage(user: Command): void {
  user
    .command("usage")
    .description("Print detailed user command documentation (LLM-optimized)")
    .action(() => {
      console.log(USAGE_TEXT.trim());
    });
}
